src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -520,6 +521,44 @@ public Task<ActionResult<IEnumerable<Models.ProcessIdentifier>>> GetProcesses()
             }, processKey, ArtifactType_Logs);
         }
 
+        /// <summary>
+        /// Gets the Dotnet Monitor Version (other configuration information may be added in the future).
+        /// </summary>
+        [HttpGet("info", Name = nameof(GetInfo))]
+        [ProducesWithProblemDetails(ContentTypes.ApplicationJson)]
+        [ProducesResponseType(typeof(IEnumerable<Models.ProcessIdentifier>), StatusCodes.Status200OK)]
+        public ActionResult<Models.DotnetMonitorInfo> GetInfo()
+        {
+            string version = GetDotnetMonitorVersion();
+
+            return this.InvokeService(() =>
+            {
+                Models.DotnetMonitorInfo dotnetMonitorInfo = new Models.DotnetMonitorInfo()
+                {
+                    Version = version
+                };
+                
+                _logger.WrittenToHttpStream();
+                return new ActionResult<Models.DotnetMonitorInfo>(dotnetMonitorInfo);
+            }, _logger);
+        }
+
+        private string GetDotnetMonitorVersion()
+        {
+            var assembly = Assembly.GetEntryAssembly() ?? Assembly.GetExecutingAssembly();
```

Comment:
Hmm. I think we want this to always be one or the other. Probably return either `Assembly.GetExecutingAssembly()` or return null.

Reply:
The way Kyle has it is correct. This is exactly what System.CommandLine does, at least the version we are using for dotnet-monitor.